src/rpc/util.cpp

```diff
@@ -1117,7 +1117,16 @@ std::string RPCArg::ToStringObj(const bool oneline) const
 
 std::string RPCArg::ToString(const bool oneline) const
 {
-    if (oneline && !m_opts.oneline_description.empty()) return m_opts.oneline_description;
+    if (oneline && !m_opts.oneline_description.empty()) {
+        if (m_opts.oneline_description[0] == '\"' && m_type != Type::STR_HEX && m_type != Type::STR && gArgs.GetBoolArg("-rpcdoccheck", DEFAULT_RPC_DOC_CHECK)) {
+            throw std::runtime_error{
+                strprintf("Internal bug detected: non-string RPC arg \"%s\" quotes oneline_description:\n%s\n%s %s\nPlease report this issue here: %s\n",
+                          m_names, m_opts.oneline_description,
+                          PACKAGE_NAME, FormatFullVersion(),
+                          PACKAGE_BUGREPORT)};
```

Comment:
The above suggestion doesn't compile without syntax changes due to ```error: unterminated argument list invoking macro "STR_INTERNAL_BUG"``` This is because of a missing ```)``` to close off the arguments for STR_INTERNAL_BUG and then there is a loose comma after ```m_opts.oneline_description```. Once these are removed code does compile but the output is not as clean as the current commit due to an extra ```"``` on a new line after the name of the ```oneline_description```

The output of ```STR_INTERNAL_BUG``` has what appears to be an extra newline with a single quote. I can take a deeper look later but a direct patch of this doesn't seem to yield a better result in my own opinion. 

![image](https://github.com/bitcoin/bitcoin/assets/3104223/f80e1c47-6a4e-4353-91e7-f79db3ea79bf)

Reply:
The above suggestion doesn't compile without syntax changes due to ```error: unterminated argument list invoking macro "STR_INTERNAL_BUG"``` This is because of a missing ```)``` to close off the arguments for STR_INTERNAL_BUG and then there is a loose comma after ```m_opts.oneline_description```. Once these are removed code does compile but the output is not as clean as the current commit due to an extra ```"``` on a new line after the name of the ```oneline_description```

The output of ```STR_INTERNAL_BUG``` has what appears to be an extra newline with a single quote. I can take a deeper look later but a direct patch of this doesn't seem to yield a better result in my own opinion. 

![image](https://github.com/bitcoin/bitcoin/assets/3104223/f80e1c47-6a4e-4353-91e7-f79db3ea79bf)

Reply:
No problem just was testing things. Compiles good but looks like a double quote on the end of the ```m_opts.oneline_description```.

![image](https://github.com/bitcoin/bitcoin/assets/3104223/97917d82-e2b0-490a-b37a-bdf79f9d68c1) 

The current PR displays correctly ```"options```

Reply:
This compiles and looks great even for the existing errors! Modifying this does cause rpc_misc.py to fail test runner. This is quite nice.